xdsl/interpreters/eqsat_pdl_interp.py

```diff
@@ -34,3 +40,60 @@ def populate_known_ops(self, module: ModuleOp) -> None:
                 self.known_ops[op] = op
             else:
                 self.eclass_union_find.add(op)
+
+    @impl(pdl_interp.GetResultOp)
+    def run_getresult(
+        self,
+        interpreter: Interpreter,
+        op: pdl_interp.GetResultOp,
+        args: tuple[Any, ...],
+    ) -> tuple[Any, ...]:
+        result = cast(
+            None | OpResult[Attribute],
+            PDLInterpFunctions.run_get_result(self, interpreter, op, args).values[0],
+        )
+
+        if result is None:
+            return (None,)
+
+        if len(result.uses) == 1 and isinstance(
+            eclass_op := next(iter(result.uses)).operation, eqsat.EClassOp
+        ):
+            assert len(eclass_op.results) == 1
+            result = eclass_op.results[0]
+        else:
+            raise InterpretationError(
+                "pdl_interp.get_result currently only supports operations with results"
+                " that are used by a single EClassOp each."
+            )
```

Comment:
I would recommend flipping this, erroring out in a nested if, and carrying on downward on the happy path:
``` python
if len(result.uses) != 1 or not isinstance(...):
    raise ...

return (eclass_op.result,)
```